docs/src/literate-tutorials/reactive_surface.jl

```diff
@@ -0,0 +1,260 @@
+#
+# ```math
+#  \begin{aligned}
+#    \partial_t r_1 &= \nabla \cdot (D_1 \nabla r_1) - r_1*r_2^2 + F *(1 - r_1) \quad \textbf{x} \in \Omega, \\
+#    \partial_t r_2 &= \nabla \cdot (D_2 \nabla r_2) + r_1*r_2^2 - r_2*(F + k ) \quad \textbf{x} \in \Omega,
+#  \end{aligned}
+# ```
+#
+# where $r_1$ and $r_2$ are the reaction fields, $D_1$ and $D_2$ the diffusion tensors,
+# $k$ is the conversion rate, $F$ is the feed rate and $\Omega$ the domain. Depending on the choice of
+# parameters a different pattern can be observed. Please also note that the domain does not have a
+# boundary. The corresponding weak form can be derived as usual.
+#
+# For simplicity we will solve the problem with the Lie-Troter-Godunov operator splitting technique with
+# the classical reaction-diffusion split. In this method we split our problem in two problems, i.e. a heat
+# problem and a pointwise reaction problem, and solve them alternatingly to advance in time.
+#
+#-
+# ## Commented Program
+#
+# Now we solve the problem in Ferrite. What follows is a program spliced with comments.
+#md # The full program, without comments, can be found in the next [section](@ref reactive_surface-plain-program).
+#
+# First we load Ferrite, and some other packages we need
+
+using Ferrite, FerriteGmsh
+using BlockArrays, SparseArrays, LinearAlgebra
+
+# ### Assembly routines
+# The following assembly routines are similar to these found in previous tutorials.
+
+function assemble_element_mass!(Me::Matrix, cellvalues::CellValues)
+    n_basefuncs = getnbasefunctions(cellvalues)
+    ## Reset to 0
+    fill!(Me, 0)
+    ## Loop over quadrature points
+    for q_point in 1:getnquadpoints(cellvalues)
+        ## Get the quadrature weight
+        dΩ = getdetJdV(cellvalues, q_point)
+        ## Loop over test shape functions
+        for i in 1:n_basefuncs
+            δuᵢ = shape_value(cellvalues, q_point, i)
+            ## Loop over trial shape functions
+            for j in 1:n_basefuncs
+                δuⱼ = shape_value(cellvalues, q_point, j)
+                ## Add contribution to Ke
+                Me[2*i-1, 2*j-1] += (δuᵢ * δuⱼ) * dΩ
+                Me[2*i  , 2*j  ] += (δuᵢ * δuⱼ) * dΩ
+            end
+        end
+    end
+    return nothing
+end
+
+function assemble_element_diffusion!(De::Matrix, cellvalues::CellValues)
+    n_basefuncs = getnbasefunctions(cellvalues)
+    ## Reset to 0
+    fill!(De, 0)
+    ## Loop over quadrature points
+    for q_point in 1:getnquadpoints(cellvalues)
+        ## Get the quadrature weight
+        dΩ = getdetJdV(cellvalues, q_point)
+        ## Loop over test shape functions
+        for i in 1:n_basefuncs
+            ∇δuᵢ = shape_gradient(cellvalues, q_point, i)
+            ## Loop over trial shape functions
+            for j in 1:n_basefuncs
+                ∇δuⱼ = shape_gradient(cellvalues, q_point, j)
+                ## Add contribution to Ke
+                De[2*i-1, 2*j-1] += 2*0.00008 * (∇δuᵢ ⋅ ∇δuⱼ) * dΩ
```

Comment:
Are these "material parameters"?

Reply:
Oh, right. That is a material parameter (diffusion coefficient). Let me fix it.